packages/frontend-2/pages/server-management/active-users.vue

```diff
@@ -0,0 +1,332 @@
+import { graphql } from '~~/lib/common/generated/gql'
+import {
+  MagnifyingGlassIcon,
+  ShieldExclamationIcon,
+  ShieldCheckIcon,
+  TrashIcon
+} from '@heroicons/vue/20/solid'
+import { UserItem } from '~~/lib/server-management/helpers/types'
+import {
+  convertThrowIntoFetchResult,
+  getFirstErrorMessage
+} from '~~/lib/common/helpers/graphql'
+
+definePageMeta({
+  middleware: ['admin']
+})
+
+const getUsers = graphql(`
+  query UserList($limit: Int!, $cursor: String, $query: String) {
+    admin {
+      userList(limit: $limit, cursor: $cursor, query: $query) {
+        totalCount
+        cursor
+        items {
+          id
+          email
+          avatar
+          name
+          role
+          verified
+          company
+        }
+      }
+    }
+  }
+`)
+
+const adminDeleteUser = graphql(`
+  mutation Mutation($userConfirmation: UserDeleteInput!) {
+    adminDeleteUser(userConfirmation: $userConfirmation)
+  }
+`)
+
+const changeRoleMutation = graphql(`
+  mutation AdminChangeUseRole($userRoleInput: UserRoleInput!) {
+    userRoleChange(userRoleInput: $userRoleInput)
+  }
+`)
+
+const userToModify = ref<UserItem | null>(null)
+const searchString = ref('')
+const showUserDeleteDialog = ref(false)
+const showChangeUserRoleDialog = ref(false)
+const newRole = ref('')
+const oldRole = computed(() => userToModify.value?.role ?? '')
+const logger = useLogger()
+const infiniteLoaderId = ref('')
+const {
+  result: extraPagesResult,
+  fetchMore: fetchMorePages,
+  variables: resultVariables,
+  onResult,
+  refetch: refetchUsers
+} = useQuery(getUsers, () => ({
+  limit: 50,
+  query: searchString.value
+}))
+const moreToLoad = computed(
+  () =>
+    !extraPagesResult.value?.admin?.userList ||
+    extraPagesResult.value.admin.userList.items.length <
+      extraPagesResult.value.admin.userList.totalCount
+)
+const users = computed(() => extraPagesResult.value?.admin.userList.items || [])
+
+const { triggerNotification } = useGlobalToast()
+const { mutate: adminDeleteUserMutation } = useMutation(adminDeleteUser)
+const { mutate: mutateChangeRole } = useMutation(changeRoleMutation)
+
+const openUserDeleteDialog = (user: UserItem) => {
+  userToModify.value = user
+  showUserDeleteDialog.value = true
+}
+
+const closeUserDeleteDialog = () => {
+  showUserDeleteDialog.value = false
+}
+
+const openChangeUserRoleDialog = (user: UserItem, newRoleValue: string) => {
+  userToModify.value = user
+  newRole.value = newRoleValue
+  showChangeUserRoleDialog.value = true
+}
+
+const closeChangeUserRoleDialog = () => {
+  showChangeUserRoleDialog.value = false
+}
+
+const deleteConfirmed = async () => {
+  if (!userToModify?.value?.email) {
+    return
+  }
+  const result = await adminDeleteUserMutation({
+    userConfirmation: { email: userToModify.value.email }
+  }).catch(convertThrowIntoFetchResult)
+  if (result?.data?.adminDeleteUser) {
+    closeUserDeleteDialog()
+    refetchUsers()
```

Comment:
Refetch is fine for now, but in the future it would be preferred to use post-mutation cache updates to just remove the deleted item from the cached list